prbt_hardware_support/src/pilz_modbus_read_client_node.cpp

```diff
@@ -39,42 +41,71 @@ int main(int argc, char **argv)
 {
   ros::init(argc, argv, "modbus_read_node");
 
-  ros::NodeHandle nh {"~"};
+  ros::NodeHandle pnh{"~"};
+  ros::NodeHandle nh;
+
+  bool has_register_range_parameters = pnh.hasParam(PARAM_NUM_REGISTERS_TO_READ_STR) &&
```

Comment:
Seems to me that you could condense all parts handling this into one section rather than distributing it all over the file